lib/iris/_merge.py

```diff
@@ -1809,7 +1809,8 @@ def key_func(coord):
 
         # Order the coordinates by hints, axis, and definition.
         for coord in sorted(coords, key=key_func):
-            if not cube.coord_dims(coord) and coord.shape == (1,):
+            dims = tuple(cube.coord_dims(coord))
+            if not dims and coord.shape == (1,):
```

Comment:
FWIW I was looking at this only yesterday, in [quite another context](https://github.com/SciTools/iris/issues/4530#issuecomment-1022220908)
And, I think that the shape **_didn't_** really need testing : 
if it has no cube-dims, it **must** be scalar, no?